src/tests/readytorun/JittedMethodsCountingTest/JittedMethodsCountingTest.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+using Reflection = System.Reflection;
+using InteropServices = System.Runtime.InteropServices;
+
+public class JittedMethodsCountingTest
+{
+    private const int MAX_JITTED_METHODS_ACCEPTED = 50;
+
+    [Fact]
+    public static int TestEntryPoint()
+    {
+        // If DOTNET_ReadyToRun is disabled, then this test ought to be skipped.
+        if (!IsReadyToRunEnvSet())
+        {
+            Console.WriteLine("\nThis test is only supported in ReadyToRun scenarios."
+                              + " Skipping...\n");
+            return 100;
+        }
+
+        if (IsRunCrossgen2Set() && IsRunningOnARM64())
+        {
+            Console.WriteLine("\nThis test is currently unsupported on ARM64 when"
+                              + " RunCrossGen2 is enabled. Skipping...\n");
+            return 100;
+        }
+
+        string testAppLocation = Path.GetDirectoryName(Reflection
+                                                      .Assembly
+                                                      .GetExecutingAssembly()
+                                                      .Location);
+        string appName = Path.Combine(testAppLocation, "HelloWorld.dll");
+
+        // For adding any new apps for this test, make sure they return a negative
+        // number when failing.
+        int appResult = RunHelloWorldApp(appName);
```

Comment:
The setup with the app running in a separate process made sense when the test used JIT logging to get the method count. 

This setup is overkill now that the test just calls `GetCompiledMethodCount` and checks its value. Would it be better to turn this into an ordinary single process test that just calls `GetCompiledMethodCount` and checks its value (and skipping the check in specific environments)?